boot/bootutil/src/loader.c

```diff
@@ -517,18 +523,22 @@ boot_check_header_erased(struct boot_loader_state *state, int slot)
 }
 
 #if (BOOT_IMAGE_NUMBER > 1) || \
+    defined(MCUBOOT_DIRECT_XIP) || \
     (defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION))
 /**
  * Check if the version of the image is not older than required.
  *
- * @param req         Required minimal image version.
- * @param ver         Version of the image to be checked.
+ * @param req           Required minimal image version.
+ * @param ver           Version of the image to be checked.
+ * @param accept_equal  If true, an equal version number is accepted, otherwise
+ *                      if false it is considered as not sufficient.
  *
- * @return            0 if the version is sufficient, nonzero otherwise.
+ * @return              0 if the version is sufficient, nonzero otherwise.
  */
 static int
 boot_is_version_sufficient(struct image_version *req,
```

Comment:
Rather than a flag `accept_equal`, I think it would be cleaner to just rename this function to something like `boot_version_cmp` that works like memcmp or strcmp. It returns -1 if the first version is strictly less, 1 if strictly greater, and 0 if the versions are equal. It can then be up to the caller to interpret that as desired. This also makes the error handling cleaner, because this function doesn't return a specific error code, and that can depend on where it is used.